src/main/java/ru/testtask/service/BackgroundJobService.java

```diff
@@ -0,0 +1,102 @@
+package ru.testtask.service;
+
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
+import ru.testtask.dto.BackgroundJobDTO;
+import ru.testtask.model.BackgroundJob;
+import ru.testtask.model.BackgroundJobStatus;
+import ru.testtask.repo.BackgroundJobRepo;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayDeque;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class BackgroundJobService {
+    @Autowired
+    private BackgroundJobRepo backgroundJobRepo;
+
+    private final ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+
+    private final ArrayDeque<BackgroundJob> backgroundJobs = new ArrayDeque<>();
+
+    private BackgroundJob currentBackgroundJob;
+
+    @PostConstruct
+    public void init(){
+        threadPoolTaskScheduler.setDaemon(true);
+        threadPoolTaskScheduler.initialize();
+    }
+
+    public void startJob(){
+        currentBackgroundJob = backgroundJobs.pop();
+        threadPoolTaskScheduler.execute(currentBackgroundJob.getJobExecution());
+    }
+
+    public void saveJob(BackgroundJob backgroundJob) {
+        backgroundJob.setJobStatus(BackgroundJobStatus.IN_PROGRESS);
+        backgroundJobs.add(backgroundJob);
+        backgroundJobRepo.save(backgroundJob);
+    }
+
+    public void createJob(String name){
+        BackgroundJob backgroundJob = BackgroundJob.builder().name(name).build();
+        backgroundJob.setJobExecution(() -> {
+            for (int i = 0; i < 10; i++) {
+                if (backgroundJob.getJobStatus().equals(BackgroundJobStatus.IN_PROGRESS)) {
```

Comment:
The job will be interrupted never because you check job status from memory but you change it in db only.